### PR TITLE
dsqss: add smoke test code

### PR DESCRIPTION
**Pull request by ketsubouchi**
Add test function.
See https://issp-center-dev.github.io/dsqss/manual/v2.0.3/en/dla/tutorial/spindimer.html

### CODE REVIEW

**Comment by ketsubouchi** [comment]
Fixed style eror

File: var/spack/repos/builtin/packages/dsqss/package.py [hunk 1]
```diff
@@ -45,3 +45,23 @@ def cmake_args(self):
         ]
 
         return args
+
+    def test(self):
+        test01 = find(self.prefix.share, '01_spindimer')[0]
+        copy(join_path(test01, 'std.toml'), '.')
+        # prepare
+        pythonexe = self.spec['python'].command.path
+        opts = [self.spec.prefix.bin.dla_pre, 'std.toml']
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dsqss/package.py:54]: This file -- `dla_pre` -- does not appear to be present when I install the package using `spack install dsqss +mpi`.  (I cannot build locally without `+mpi` due to `numactl` configuration errors for *any* of its versions.)

File: var/spack/repos/builtin/packages/dsqss/package.py [hunk 2]
```diff
@@ -45,3 +45,23 @@ def cmake_args(self):
...
+        # (mpi) run
+        opts = []
+        if self.spec.satisfies('+mpi'):
+            exe_name = self.spec['mpi'].prefix.bin.mpirun
+            opts.extend(['-n', '1'])
+            opts.append(join_path(self.prefix.bin, 'dla'))
+        else:
+            exe_name = 'dla'
+        opts.append('param.in')
+        expected = ['R ene = -3.74300000e-01 2.96344394e-03']
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/dsqss/package.py:65]: Is this the only text output to `sample.log`?

**Comment by ketsubouchi** [comment]
Thank you for your kind suggestion
1. dla_pre
`dla_pre` is Python script wrapper.
The make process of DSQSS downloads python files such as https://bootstrap.pypa.io/get-pip.py and build `dla_pre`.
So please try to export `http_proxy` and `https_proxy`

In my environment (x86 VMware + CentOS7 + gcc8.3.0(by spack)), ~mpi also succeed. 

2. Is this the only text output to sample.log?
No. please see installed reference (share/dsqss/2.0.3/samples/dla/01_spindimer/ref/sample.log). 
However, `sample.log` does not include message like "Finished" or "All done.".
So I checked only `R ene` line. Because it was checked in the tutorial.

File: var/spack/repos/builtin/packages/dsqss/package.py (hunk 1 above)
Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/dsqss/package.py:54]: `dla_pre` is Python script wrapper.
The make process of DSQSS downloads python files such as https://bootstrap.pypa.io/get-pip.py and build dla_pre.
So please try to export `http_proxy` and `https_proxy`

File: var/spack/repos/builtin/packages/dsqss/package.py (hunk 2 above)
Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/dsqss/package.py:65]: No. please see installed reference (share/dsqss/2.0.3/samples/dla/01_spindimer/ref/sample.log).
However, sample.log does not include message like "Finished" or "All done.".
So I checked only return code and `R ene` line. 
Only this line was checked in the tutorial.

File: var/spack/repos/builtin/packages/dsqss/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/dsqss/package.py:54]: The package installation and test processes should be self contained.  By that I mean the user should be able to run:

```
$ spack install dsqss
$ spack test run dsqss
```

without having to do anything beyond setting up the Spack environment.

File: var/spack/repos/builtin/packages/dsqss/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/dsqss/package.py:65]: That's fine.  I was just wondering.

File: var/spack/repos/builtin/packages/dsqss/package.py (hunk 1 above)
Comment (ketsubouchi) [reply var/spack/repos/builtin/packages/dsqss/package.py:54]: Please share `spack-build-out.txt`.
Is there error like this?

> WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x40003989ce50>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pip/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4000398b9ca0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pip/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4000398b9fd0>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pip/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x4000398b9c70>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pip/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x40003953e100>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pip/
ERROR: Could not find a version that satisfies the requirement pip (from versions: none)
ERROR: No matching distribution found for pip

**Comment by a-saitoh-fj** [comment]
@tldahlgren 
In order to change the person in charge, I will end this PR and start a new PR.
In this PR, Ketsubouchi requested a spack log. The reason is that the DSQSS make process downloads a Python file such as https://bootstrap.pypa.io/get-pip.py and builds dla_pre. He thought he couldn't make dla_pre because of the download.

In the new PR, remove that download and use depends_on.
Sorry, please check DSQSS again with the new PR.

**Pull request closed without merging**